### ISSUE TITLE
Error response from API when sending underscores

### ISSUE THREAD

**Issue by Pitasi** [eternnoir/pyTelegramBotAPI#114]
Hi there!
Thanks for your cool API, really easy to use.

I found a problem when trying to send messages that contains underscores.
I investigated a bit and I saw that:
https://api.telegram.org/botMYTOKEN/sendMessage?parse_mode=markdown&chat_id=SOMECHATID&text=Just_making_some_tests
will NOT work.

The solution is quite easy, just escape any _ character like this:
https://api.telegram.org/botMYTOKEN/sendMessage?parse_mode=markdown&chat_id=SOMECHATID&text=Just\_making\_some\_tests
and it will work without issues.

Basically, in my bot I just made `str.replace("_", "\_")` to make it works, but I think it should be fixed in the module.

Thanks again, and have a nice day.
Antonio

_(I'm a noob here at GitHub and with git in general, hope I'm writing in the right place :))

**UPDATE:** This issue happens only when you set `parse_mode='markdown'` sending the message and Telegram can't find a match for some _.
Now it does make sense.

**Comment by Kondra007** [comment]
Just to make it all clear to anyone reading this issue:

`Just_making_some_tests` contains odd number of underscores, so Telegram API request will fail with parse error. When using `parse_mode` "Markdown" or "HTML", you **must** check validity of text.
